File: move_zeros.rb
# @param {Integer[]} nums
# @return {Void} Do not return anything, modify nums in-place instead.
def move_zeroes(nums)
  nonzero_idx = 0
  nums.each_with_index do |n, idx|
      if n != 0
          nums[idx], nums[nonzero_idx] = nums[nonzero_idx], nums[idx]
          nonzero_idx += 1
      end
  end
  nums
end

File: move_zeros.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
  let non_zero_idx = 0;
  for (let i = 0; i <= nums.length - 1; i++) {
    if (nums[i] != 0) {
      [nums[i], nums[non_zero_idx]] = [nums[non_zero_idx], nums[i]];
      non_zero_idx++;
    }
  }
  return nums;
};
